### ISSUE TITLE
aalib package

### ISSUE THREAD

**Issue by Jehan** [msys2/MINGW-packages#11115]
In GIMP, we use the library `aalib` for ascii-art export. Here is our build rules for Windows: https://gitlab.gnome.org/GNOME/gimp/-/blob/master/build/windows/gitlab-ci/build-deps-msys2.sh#L54

As you can see, there are 2 patches, also we update manually some `config.*` files because the ones in the tarball were too old and not recognizing some modern platforms, which also means we need to regenerate the autotools files afterwards.

It's explained in this commit message: https://gitlab.gnome.org/GNOME/gimp/-/commit/62ae32123a12492926bfcc2563ee780567b46f4b

Now we haven't had any problem with it and it's not like it's taken much time in our build. But if MSYS2 wanted to integrate this as a package, it would still be easier for us. :-)

**Comment by Biswa96** [comment]
Hi @Jehan, Thank you for reporting the issue. I have added a pull request to fix this issue. Would you like to test the packages from GitHub Actions artifacts of that pull request? The test packages can be downloaded from here https://github.com/msys2/MINGW-packages/actions/runs/2054503927

**Comment by Jehan** [comment]
Sure. Could you just explain me how to install these? I see I can download a `MINGW64-packages.zip` (which I guess is what we want). Is that it? Where should I unzip it in our windows CI? Or can pacman install from this?

**Comment by jeremyd2019** [comment]
inside MINGW-packages.zip are pacman packages (actually just one in this case).  after unzipping you could do something like `pacman -U *.pkg.tar.*`

**Comment by lazka** [comment]
(if you're only testing via CI, we also just merge it and you can test once it's in the repo)

**Comment by Jehan** [comment]
@lazka Yes I am only testing through CI, then check the log to see it were correctly detected, and would finally test the CI-generated installer in a Windows VM.

I'm not doing a full install of however-this-all-MSYS2-on-Windows thing work, and a build of GIMP in a Windows VM, for sure. 😜

Anyway I can try to `pacman` in a feature branch in the CI tomorrow, either from this zip file, or from the main package if you push it (which obviously would be simpler). Just tell me. :-)

**Comment by lazka** [comment]
@Jehan -> https://packages.msys2.org/base/mingw-w64-aalib

**Comment by Jehan** [comment]
CI running! I'll comment with result in a few hours, I guess. :-)

**Comment by Biswa96** [comment]
> CI running! I'll comment with result in a few hours

![image](https://c.tenor.com/13VnwKt5qS0AAAAM/waiting.gif)

**Comment by Jehan** [comment]
All good, I now checked the log: libaa is well detected by the configure. Then I tested in a VM but it tells me that libaa-1.dll is missing. So I can't export images as ASCII art. Though it might just be the packaging script which is failing us. I'll have to look at logs in more detail.

@Biswa96 Ahah yes, it takes some time. 😉 The full Windows pipeline took 71 minutes. It's actually not too bad, considering it contains both the 32 and 64-bit Windows build, full-featured and from scratch (babl and GEGL as still self-built too, not taken from MSYS2), as well as the whole packaging and installer build (which used to take more than a hour just for this stage not so long ago until we optimized some part of the packaging! Now it takes like 10/15 minutes!) and that we still use autotools (which is still our official build system, despite meson being present).
